### PR TITLE
Rework of test scan results

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#1408|

This PR includes a set of changes for stabilization purpose as:
- Use always Wazuh-DB to query databases as long it was possible
- Don't stop the Wazuh-DB service in the decorator function 
- Databases setup improvement
- Add `insert_osinfo` otherwise the scan will fail
- Queries refactor and improvements

Also implementation of new tests as: 
- Vulnerabilities insertion checking the alerts.json file
- Vulnerabilities removal checking the ossec.conf log file
- Vulnerabilities removal checking the alerts.json file
- Vulnerabilities existence and non-existence query the vulnerabilities inventory database directly
- Testing for discarded kernel packages 

## Tests

- [X] Proven that tests **pass** when they have to pass.
- [X] Proven that tests **fail** when they have to fail.

- [X] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [X] Python codebase is documented following the Google Style for Python docstrings.
- [X] The test is documented in wazuh-qa/docs.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/vulnerability_detector.py [hunk 1]
```diff
@@ -937,10 +1007,14 @@ def create_mocked_agent(name="centos8-agent", ip="127.0.0.1", register_ip="127.0
...
 
-    # Copy agent db to create a new one
-    file.copy(os.path.join(DB_PATH, '000.db'), os.path.join(DB_PATH, f"{agent_id_str}.db"))
-
-    make_query(GLOBAL_DB_PATH, [query_create_agent])
+    # Restarting Wazuh-DB before creating new DB
+    control_service('stop', daemon='wazuh-db')
+    sleep(3)
+    control_service('start', daemon='wazuh-db')
+    sleep(3)
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply deps/wazuh_testing/wazuh_testing/vulnerability_detector.py]: Thanks Jonathan. Indeed it was not necessary, it must have been a WA when the tests were not stable.

**Comment by DProvinciani** [comment]
### Jenkins execution

https://ci.wazuh.info/job/Test_integration/6552/

image

File: deps/wazuh_testing/wazuh_testing/vulnerability_detector.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DProvinciani) [reply deps/wazuh_testing/wazuh_testing/vulnerability_detector.py:1016]: I spent some time analyzing whether or not this is needed. The thing is that considering that the agent database now is not created by the test (it is done by calling Wazuh DB), we should restart the service to allow the creation.

We verified that without this logic, the tests fail in the execution.

**Review by jmv74211 — APPROVED** [review]

File: deps/wazuh_testing/wazuh_testing/vulnerability_detector.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply deps/wazuh_testing/wazuh_testing/vulnerability_detector.py]: Yes, thanks. I didn't think about the existence of a restart option.
image

**Pull request merged**